### PR TITLE
Hyper Hype Beast Refactoring 4 Pull request

### PR DESCRIPTION
**Pull request by arthurysong**
I refactored Beef.Users logic into the subdirectories created in Beef (access, mutations, and queries). @ityonemo , can you take a look at the pull Request and see if it makes sense?

### CODE REVIEW

**Comment by benawad** [comment]
Are you we doing `Beef.Users` or `Beef.Mutations.User`? Or is `Beef.Users` just a middle step?

**Comment by ityonemo** [comment]
hold on this, I haven't looked through it yet, sorry.

**Comment by ityonemo** [comment]
@arthurysong I couldn't figure out how to pr into your "pullrequest" branch.
- Good job on getting ahead of me in terms of moving code to the private modules
- I think there might have been a misunderstanding about best practices, but what I was planning with the Users and the defdelegates should be clear now.
- Note the role of the Queries module and the example for: Access.Users.find_by_github_ids.

here is an updated pr:
https://github.com/ityonemo/dogehouse/tree/hyperhypebeast-refactoring-4-update

Now that you've seen the template of how to do do things it should be easy to moving more queries out to the Queries module or moving out content from the other context modules into their Mutations and Access private modules.

**Comment by ityonemo** [comment]
I recommend merging my branch into yours and submitting under this PR =D

**Comment by ityonemo** [comment]
also @benawad note that Beef.Mutations.Users contains a few boundary violations, (it calls Gen modules) so in the long run i'll be looking to refactor those.

File: kousa/lib/socket_handler.ex [hunk 1]
```diff
@@ -543,8 +544,8 @@ defmodule Kousa.SocketHandler do
   end
 
   def handler("ask_to_speak", _data, state) do
-    with {:ok, room_id} <- Users.tuple_get_current_room_id(state.user_id) do
-      case Kousa.Data.RoomPermission.ask_to_speak(state.user_id, room_id) do
+    with {:ok, room_id} <- Beef.Queries.User.tuple_get_current_room_id(state.user_id) do
+      case RoomsPermission.ask_to_speak(state.user_id, room_id) do
```
Comment (Choongkyu) [inline kousa/lib/socket_handler.ex]: where is the `RoomsPermission` module defined?  [This pull request](https://github.com/benawad/dogehouse/pull/543) made the identical change which I tried to [address here](https://github.com/benawad/dogehouse/pull/581).

**Comment by arthurysong** [comment]
Oh wow, that makes a lot more sense . Thanks a lot for the update and the feedback. Yeah, I think I have a better idea about how to move break down the code into smaller queries. I was a bit confused at first about what "composable queries" meant. xD. Yeah, I'll merge your updated hhb4 update and continue refactoring in that manner.

**Comment by arthurysong** [comment]
The defdelegates make it so much nicer too :)

**Comment by arthurysong** [comment]
@benawad @ityonemo I updated this pull request with the updates ityonemo made to his fork: the context module Beef.User is properly being used and gives access to the Beef.User.mutation and Beef.User.access functions. I also started refactoring more of the users.mutation methods to use smaller query functions.

**Comment by ityonemo** [comment]
@arthurysong you're doing a fantastic job.  Only further comment is consider in the future to name your branch something more meaningful!  @benawad This looks good, I'll do stage five starting from where this leaves off.

**Comment by ityonemo** [comment]
@arthurysong can you rebase against latest staging and fix the merge conflicts?

**Comment by arthurysong** [comment]
@ityonemo okay, I rebased using the latest staging and fixed all the merge conflicts

**Comment by ityonemo** [comment]
:tada: @benawad

@arthurysong If you want to work on refactoring the other modules to be like User, that can be HHB-6, I'll also do a PR where I show you the next stage of the DB modules (and it will teach you the basics of macros), which can be HHB-8.  I don't advocate overusing macros, though, so I'll be sure to add in a comment justification for these macros.  In the meantime I'll do HHB-7 which is some work moving Kousa.BL to "Zuke"

**Comment by arthurysong** [comment]
@ityonemo awesome! ill start working on hhb-6 for the other modules. yeah I've been trying to understand what macros are but they're a bit confusing :P

**Pull request merged**